crates/meilisearch-auth/src/store.rs

```diff
@@ -345,3 +326,35 @@ where
     let head = head.try_into().ok()?;
     Some((head, tail))
 }
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+struct KeyActions {
+    actions: u64, // Bitflags for actions
+    indexes: Vec<IndexUidPattern>,
+    expires_at: Option<OffsetDateTime>,
+}
+
+impl KeyActions {
+    fn new(
+        actions: &HashSet<Action>,
+        indexes: &[IndexUidPattern],
+        expires_at: Option<OffsetDateTime>,
+    ) -> Self {
+        let mut bitflags = 0u64;
+        for (index, action_flag) in enum_iterator::all::<Action>().enumerate() {
+            if actions.contains(&action_flag) {
+                bitflags |= 1 << index;
+            }
+        }
```

Comment:
I don't think you need to enumerate all the Actions to retrieve their bitflags, something like the below code should be possible: 
```rust
        for action in actions {
            bitflags |= action.bits();
        }
```


```suggestion
        for action in actions {
            bitflags |= action.bits();
        }
```

Reply:
if we are removing actions than how we can do this?? I have almost implemented bit masks for this file trying to fix one error I will do a commit then tell me if something is wrong or not